compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -609,7 +610,7 @@ fn find_opaque_ty_constraints(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Ty<'_> {
             // ```
             let tables = self.tcx.typeck(def_id);
             if let Some(_) = tables.tainted_by_errors {
-                self.found = Some((DUMMY_SP, self.tcx.ty_error()));
+                self.found = Some(ty::OpaqueHiddenType { span: DUMMY_SP, ty: self.tcx.ty_error() });
```

Comment:
I hope we correctly silence these 😅

I was about to tell you off for the `DUMMY_SP` but that's of course preexisting 😊